### PR TITLE
Refactor PolarSeries and cleanup PieSeries.

### PR DESCRIPTION
**Pull request by alantreadway**
Refactors:
- Common selectors and groups up from radial series to `PolarSeries`.
- Applies the use of `fromToMotion()` to `PieSeries`.

### CODE REVIEW

File: packages/ag-charts-community/src/chart/polarChart.ts [hunk 1]
```diff
@@ -80,7 +79,7 @@ export class PolarChart extends Chart {
     }
 
     private async computeCircle(seriesBox: BBox) {
-        const polarSeries = this.series.filter((series): series is PolarSeries<SeriesNodeDatum> => {
+        const polarSeries = this.series.filter((series): series is PolarSeries<any, any> => {
```
Comment (lsjroberts) [inline packages/ag-charts-community/src/chart/polarChart.ts:82]: any?

File: packages/ag-charts-community/src/chart/series/polar/pieSeries.ts [hunk 2]
```diff
@@ -879,43 +864,43 @@ export class PieSeries extends PolarSeries<PieNodeDatum> {
...
+            sector.stroke = format.stroke;
+            sector.strokeWidth = format.strokeWidth!;
+            sector.fillOpacity = format.fillOpacity!;
             sector.strokeOpacity = this.strokeOpacity;
             sector.lineDash = this.lineDash;
             sector.lineDashOffset = this.lineDashOffset;
             sector.fillShadow = this.shadow;
             sector.lineJoin = 'round';
-            sector.visible = this.seriesItemEnabled[index];
+            // sector.visible = this.seriesItemEnabled[index];
```
Comment (lsjroberts) [inline packages/ag-charts-community/src/chart/series/polar/pieSeries.ts]: remove?

File: packages/ag-charts-community/src/chart/series/polar/pieSeries.ts [hunk 3]
```diff
@@ -1572,258 +1557,75 @@ export class PieSeries extends PolarSeries<PieNodeDatum> {
...
-            } else if (diff.removedIndices[ri] === i) {
-                sortedDatumIds.push(diff.removed[ri++]);
-            } else if (diff.updatedIndices[ui] === i) {
-                sortedDatumIds.push(diff.updated[ui++]);
-            }
+        if (diff?.removed?.length > 0 && itemSelection.hasGarbage()) {
+            // Short-circuit in-progress animation, as stacked removals
+            // don't animate very well.
+            // itemSelection.cleanup();
+            // super.resetAllAnimation();
```
Comment (lsjroberts) [inline packages/ag-charts-community/src/chart/series/polar/pieSeries.ts]: and here

**Review by lsjroberts — APPROVED** [review]
LGTM outside of minor bits

File: packages/ag-charts-community/src/chart/polarChart.ts (hunk 1 above)
Comment (alantreadway) [reply packages/ag-charts-community/src/chart/polarChart.ts:82]: Unfortunately needed as there's no way to derive it, and `never` and `unknown` cause issues later in this function :-(

File: packages/ag-charts-community/src/chart/series/polar/pieSeries.ts (hunk 3 above)
Comment (alantreadway) [reply packages/ag-charts-community/src/chart/series/polar/pieSeries.ts]: Fixed.

File: packages/ag-charts-community/src/chart/series/polar/pieSeries.ts (hunk 2 above)
Comment (alantreadway) [reply packages/ag-charts-community/src/chart/series/polar/pieSeries.ts]: Fixed

File: packages/ag-charts-community/src/chart/series/cartesian/barUtil.ts [hunk 4]
```diff
@@ -144,7 +144,7 @@ export function prepareBarAnimationFunctions(startPositionFn: StartingPositionFn
     const fromFn = (rect: Rect, datum: AnimatableBarDatum, status: NodeUpdateState) => {
         if (status === 'removed') {
             return { x: datum.x, y: datum.y, width: datum.width, height: datum.height };
-        } else if (status === 'added') {
+        } else if (status === 'added' || status === 'unknown') {
```
Comment (alexanderby) [inline packages/ag-charts-community/src/chart/series/cartesian/barUtil.ts:147]: image

File: packages/ag-charts-community/src/scene/selection.ts [hunk 5]
```diff
@@ -125,6 +132,14 @@ export class Selection<TChild extends Node = Node, TDatum = any> {
         return this;
     }
 
+    isGarbage(node: TChild) {
```
Comment (alexanderby) [inline packages/ag-charts-community/src/scene/selection.ts:135]: image

**Review by alexanderby — APPROVED** [review]

**Pull request merged**